src/indicator/momentum/ichimokuCloud.ts

```diff
@@ -59,37 +123,26 @@ export const IchimokuCloudDefaultConfig: Required<IchimokuCloudConfig> = {
  * @return ichimoku cloud result object.
  */
 export function ichimokuCloud(
-  highs: number[],
-  lows: number[],
-  closings: number[],
-  config: IchimokuCloudConfig = {}
+    highs: number[],
+    lows: number[],
+    closings: number[],
+    config: IchimokuCloudConfig = {}
 ): IchimokuCloudResult {
-  checkSameLength(highs, lows, closings);
+    checkSameLength(highs, lows, closings);
+
+    const {short, medium, long, close} = {
+        ...IchimokuCloudDefaultConfig,
+        ...config,
+    };
 
-  const { short, medium, long, close } = {
-    ...IchimokuCloudDefaultConfig,
-    ...config,
-  };
-  const conversion = divideBy(
-    2,
-    add(mmax(highs, { period: short }), mmin(lows, { period: short }))
-  );
-  const base = divideBy(
-    2,
-    add(mmax(highs, { period: medium }), mmin(lows, { period: medium }))
-  );
-  const leadingSpanA = divideBy(2, add(conversion, base));
-  const leadingSpanB = divideBy(
-    2,
-    add(mmax(highs, { period: long }), mmin(lows, { period: long }))
-  );
-  const laggingSpan = shiftRightBy(close, closings);
+    const tenkanSen = calculateTenkanSen({highs, lows, short})
+    const kijunSen = calculateKijunSen({highs, lows, medium})
 
-  return {
-    conversion,
-    base,
-    leadingSpanA,
-    leadingSpanB,
-    laggingSpan,
-  };
+    return {
+        conversion: tenkanSen,
+        base: kijunSen,
+        leadingSpanA: calculateSenkouSpanA({tenkanSen, kijunSen, medium}),
```

Comment:
I wonder if we should keep a single naming. Do you know if the Senkou Span or Leading Span is much better known? Perhaps then we can name both the object fields and the method names the same.

Reply:
Tenkan, Kijun, SSA, SSB, and LagginSpan are widely used among people working with Ichimoku.
I didn't want to introduce a "breaking change" in the code base, that's why I didn't touch `IchimokuCloudResult`
If you think it's to change`IchimokuCloudResult` interface then it could be
```ts
export interface IchimokuCloudResult {
    kijun: number[];
    tenkan: number[];
    ssa: number[];
    ssb: number[];
    laggingSpan: number[];
}
```